queries.sql

```diff
@@ -6,3 +6,42 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT neutered, MAX(escape_attempts) as escape_attempts FROM animals
+ GROUP BY neutered ORDER BY MAX(escape_attempts) DESC;
+SELECT neutered, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY neutered;
+SELECT species, MAX(weight_kg) as max_weight , MIN(weight_kg) as min_weight FROM animals
+ GROUP BY species;
+SELECT species, AVG(escape_attempts) as avg_escape_attempts FROM animals
+WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY species;
+SELECT neutered, AVG(escape_attempts) as avg_escape_attempts FROM animals
+ WHERE date_of_birth >= '1990-01-01' AND  date_of_birth < '2000-01-01' GROUP BY neutered;
```

Comment:
- [ ] Kindly add the query for `What is the average weight of animals?`, you have it in the screenshots you have  provided, but I do not see it here.
Your screenshot
![image](https://github.com/torobucci/vet_clinic/assets/81932478/702a83a6-2b64-423d-8e4d-c4d508f0394a)

Reply:
Update queries.sql